plugins/star-rating/tests.js

```diff
@@ -7,55 +7,255 @@ var API_KEY_ADMIN = "";
+            .end(function(err, res) {
+            	if (err) return done(err);
+            	var ob = JSON.parse(res.text);
+            	ob.should.have.property('result','Success');
+            	setTimeout(done, 100);
+            });
+        });
+    });
+	
+	describe('Creating event', function() {
+        it('should success', function(done) {
+            API_KEY_ADMIN = testUtils.get("API_KEY_ADMIN");
+            APP_ID = testUtils.get("APP_ID");
+            APP_KEY = testUtils.get("APP_KEY");
+            WIDGET_ID = testUtils.get("WIDGET_ID");
+
+            var events = [{
+                "key": "[CLY]_star_rating",
+                "count": 1,
+                "sum": 1,
+                "segmentation": {
+                	"contactMe":true,
+                	"email":"fb@count.ly",
+                	"comment":"It's a test comment.",
+                    "rating": 5,
+                    "app_version": "1.23",
+                    "platform": "iOS",
+                    "widget_id": WIDGET_ID
+                }
+            }];
+
+            request.get('/i?app_key=' + APP_KEY + '&device_id=' + 1 + "&events=" + JSON.stringify(events))
+                .expect(200)
+                .end(function(err, res) {
+                    if (err) return done(err);
+                    var ob = JSON.parse(res.text);
+                    ob.should.have.property('result', 'Success');
+                    setTimeout(done, 100)
+            });
+        });
+    });
+    
+    describe('Removing widget', function() {
+        it('should success', function(done) {
+            API_KEY_ADMIN = testUtils.get("API_KEY_ADMIN");
+            APP_ID = testUtils.get("APP_ID");
+            WIDGET_ID = testUtils.get("WIDGET_ID");
+            request.get('/i/feedback/widgets/remove?api_key=' + API_KEY_ADMIN + '&app_id=' + APP_ID + '&widget_id=' + WIDGET_ID)
+            .expect(200)
+            .end(function(err, res) {
+            	if (err) return done(err);
+            	var ob = JSON.parse(res.text);
+            	ob.should.have.property('result','Success');
+            	setTimeout(done, 100);
+            });
+        });
+    });
+    
+    describe('Verify rating', function() {
         it('should return 200 for request platform info', function(done) {
-            APP_ID = testUtils.get("APP_ID") || APP_ID ;
+            APP_ID = testUtils.get("APP_ID") || APP_ID;
             var data;
-            request.get('/o?method=star&period=60days&api_key='+API_KEY_ADMIN+'&app_id='+APP_ID)
-            .end(function(err, res) {
+            request.get('/o?method=star&period=60days&api_key=' + API_KEY_ADMIN + '&app_id=' + APP_ID).end(function(err, res) {
                 res.statusCode.should.equal(200);
                 data = JSON.parse(res.text);
-                console.log(data);
                 should(data.iOS && data.iOS.indexOf('1:23') >= 0).equal(true);
                 done();
             });
         });
     });
     
```

Comment:
I would want 2 more things here
1) create another widget but do not remove it, so we could check if resetting app also removes widgets
2) Create rating/feedback without widget, to ensure backwards compatibility with old approach for older SDKs